activejob/lib/active_job/instrumentation.rb

```diff
@@ -1,6 +1,18 @@
 # frozen_string_literal: true
 
 module ActiveJob
+  class << self
+    private
+      def instrument_enqueue_all(queue_adapter, jobs)
+        payload = { adapter: queue_adapter, jobs: jobs }
+        ActiveSupport::Notifications.instrument("enqueue_all.active_job", payload) do
```

Comment:
Having a different event for mass enqueue might be a bit of a problem. e.g. tracing framework might not handle both etc.

I don't have a solution in mind right now, but making a note to put more thoughts into this.

Reply:
The advantage of a separate event is that the semantics of the current event remain the same, and frameworks can add support for the new event over time. If there is a clean way of plugging in the existing monitoring, I'd be all for that, but I'm not sure there is.

Reply:
That new event is really the only thing that bothers me here. @rafaelfranca I'd love to hear your thoughts on it?

Reply:
@Mangara ok, sorry for the delay, I just had a chat with @rafaelfranca and introducing that new event is fine, we'll just have to make sure it's properly pointed out in the changelog etc.

So no more blocker from me. I'll merge this next week.

Reply:
We might want to add this to the instrumentation guide:
https://edgeguides.rubyonrails.org/active_support_instrumentation.html#active-job

I'm happy to do this after this PR lands though, not a blocker!

Reply:
Good point! It should definitely be documented.

Reply:
Done in [3748385](https://github.com/rails/rails/commit/3748385d91bbf60edaa1af3c678cc9702f1abad0) :pray: